x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1113,12 +1125,23 @@ public void fillCacheRange(
                     SourceInputStreamFactory streamFactory,
                     int relativePos,
                     int length,
-                    IntConsumer progressUpdater
+                    IntConsumer progressUpdater,
+                    ActionListener<Void> completion
```

Comment:
nit: Since this interface is heavily used as a lambda I would prefer to name this parameter `completionListener`